src/python/gudhi/barycenter.py

```diff
@@ -0,0 +1,342 @@
+            return np.column_stack([np.zeros(m+1, dtype=int), np.arange(m+1, dtype=int)]) # TO BE CORRECTED
+    elif Y.size == 0: # X is not empty but Y is empty
+        return np.column_stack([np.zeros(n+1, dtype=int), np.arange(n+1, dtype=int)])  # TO BE CORRECTED
+
+    # we know X, Y are not empty diags now
+    M = _build_dist_matrix(X, Y)
+
+    a = np.full(n+1, 1. / (n + m) )  # weight vector of the input diagram. Uniform here.
+    a[-1] = a[-1] * m                # normalized so that we have a probability measure, required by POT
+    b = np.full(m+1, 1. / (n + m) )  # weight vector of the input diagram. Uniform here.
+    b[-1] = b[-1] * n                # so that we have a probability measure, required by POT
+    P = ot.emd(a=a, b=b, M=M)*(n+m)
+    # Note : it seems POT return a permutation matrix in this situation,
+    # ...guarantee...?
+    # It should be enough to check that the algorithm only iterates on vertices of the transportation polytope.
+    P[P < 0.5] = 0  # dirty trick to avoid some numerical issues... to be improved.
+    # return the list of (i,j) such that P[i,j] > 0, i.e. x_i is matched to y_j (should it be the diag). 
+    res = np.nonzero(P)
+    return np.column_stack(res)
+
+
+def _mean(x, m):
+    """
+    :param x: a list of 2D-points, off diagonal, x_0... x_{k-1}
+    :param m: total amount of points taken into account, that is we have (m-k) copies of diagonal
+    :returns: the weighted mean of x with (m-k) copies of the diagonal
+    """
+    k = len(x)
+    if k > 0:
+        w = np.mean(x, axis=0)
+        w_delta = _proj_on_diag(w)
+        return (k * w + (m-k) * w_delta) / m
+    else:
+        return np.array([0, 0])
+
+
+def lagrangian_barycenter(pdiagset, init=None, verbose=False):
+    """
+        Compute the estimated barycenter computed with the algorithm provided
+        by Turner et al (2014).
+        It is a local minima of the corresponding Frechet function.
+    :param pdiagset: a list of size N containing numpy.array of shape (n x 2) 
+                        (n can variate), encoding a set of 
+                        persistence diagrams with only finite coordinates. 
+    :param init: The initial value for barycenter estimate. 
+                    If None, init is made on a random diagram from the dataset. 
+                    Otherwise, it must be an int (then we init with diagset[init])
+                    or a (n x 2) numpy.array enconding a persistence diagram with n points.
+    :param verbose: if True, returns additional information about the
+                        barycenters (assignment and energy).
+    :returns: If not verbose (default), a numpy.array encoding
+                    the barycenter estimate (local minima of the energy function). 
+                    If verbose, returns a triplet (Y, a, e)
+                    where Y is the barycenter estimate, a is the assignments between the
+                    points of Y and thoses of the diagrams, 
```

Comment:
How are those assignments encoded? Essentially, how do I read `a`?

Reply:
Good question, I will detail the docs accordingly.